packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -138,8 +146,20 @@ export class MediaUpload extends React.Component {
 	}
 
 	onPickerPresent() {
+		const { autoOpenMediaUpload } = this.props;
+		const isIOS = Platform.OS === 'ios';
+
 		if ( this.picker ) {
-			this.picker.presentPicker();
+			// the delay below is required because on iOS this action sheet gets dismissed by the close event of the Inserter
+			// so this delay allows the Inserter to be closed fully before presenting action sheet.
+			if ( autoOpenMediaUpload && isIOS ) {
+				delay(
```

Comment:
That doesn't look like the most stable approach. I wonder what's that close event you're talking about? Maybe we could fix the underlying issue?

Reply:
Thanks for pointing this out :) What I meant to say here is that when the Inserter is being closed and this media upload action sheet/modal  is being shown at the same time it is closed as it opens. What I think is happening is whatever mechanism is being used internally by iOS to close the inserter modal also closes any other modals that are on the screen. So yes, I know this approach isn't stable but the slight delay just ensured the the new modal wouldn't be on the screen at the same time. Let me know what your thoughts are on this. I could investigate another approach that allows this component to somehow listen to the events of the bottom sheet. I am seeing where a `ContextProvider` is being utilized by the bottom sheet to share events so this could be extended. For more context see this [here](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/mobile/bottom-sheet/bottom-sheet-context.native.js)

Reply:
I just realized that the blocks of the block list are in a different component tree from the Inserter's bottom sheet and the bottom sheet's context provider only allows its children to consume its events so I am not sure if the approach described above would work. Is there any other way to react to propagate events across components?